godot-core/src/builtin/quaternion.rs

```diff
@@ -130,70 +138,115 @@ impl Quaternion {
         Quaternion::new(v.x, v.y, v.z, 0.0)
     }
 
+    /// Normalizes the quaternion.
+    ///
+    /// # Panics
+    /// If the quaternion has length of 0.
     pub fn normalized(self) -> Self {
-        self / self.length()
+        let length = self.length();
+        assert!(!length.approx_eq(&0.0), "Quaternion has length 0");
+        self / length
     }
 
+    /// Performs slerp
+    ///
+    /// # Panics
+    /// If either quaternion is not normalized
     pub fn slerp(self, to: Self, weight: real) -> Self {
```

Comment:
Abandoned the hand-rolled solution when I discovered that it didn't handle non-normalized vectors by erroring as it should.